### ISSUE TITLE
API for adding cards fails on Android 7 (Flyme 7.0.3.1A)

### ISSUE THREAD

**Issue by vi** [ankidroid/Anki-Android#5114]
###### Reproduction Steps

1. Install AnkiDroid on Meizu 15 Plus
2. Build and install the [API sample](https://github.com/ankidroid/apisample)
3. Manually change "Read and write to the AnkiDroid database" from Ask to Allow, both for AnkiDroid itself and for the API sample.
4. Ensure API is enabled in AnkiDroid settings.
4. Try importing something

###### Expected Result

Something is imported

###### Actual Result

After long-ish delay the API sample crashes.

    11-10 13:49:42.340 E/AndroidRuntime(32669): java.lang.IllegalArgumentException: Unknown URL content://com.ichi2.anki.flashcards/decks

My own app (TextImporter) gets `null` from `getDeckList` call (immediately if API is turn off in AnkiDroid's advanced settings, after delay if it is enabled).

###### Debug info

`ankidroid version = 2.8.4 android version = 7.0 acra`

###### Research

[x] I have read the [support page](https://ankidroid.org/docs/help.html) and am reporting a bug or enhancement request specific to AnkiDroid

[ ] I have checked the [manual](https://ankidroid.org/docs/manual.html) and the [FAQ](https://github.com/ankidroid/Anki-Android/wiki/FAQ) and could not find a solution to my issue

[ ] I have searched for similar existing issues here and on the user forum

**Comment by mikehardy** [comment]
Try installing ankidroid and allowing permissions then installing the  sample. We have had a similar issue solved this wsy. No idea why

**Comment by vi** [comment]
Do you mean uninstalling, then installing?

**Comment by mikehardy** [comment]
Yes, the previous problem we saw was that if ankidroid was not installed and I think with permission enabled before the API consumer was installed, it did not work. We don't know why as my read on manifests indicates we have the necessary declarations in ankidroid

**Comment by vi** [comment]
Uninstalled AnkiDroid, installed it again from F-Droid, launched, accepted one permission (access storage).

Both API sample and my app started working.

**Comment by mikehardy** [comment]
Sounds like the same thing we saw with the previous issue. We are not able to fix this that I know of as it appears to be a quirk with permissions. Our recommendation for the developer of the other application was to include this information prominently in their documentation - that AnkiDroid needed to be fully installed prior to installing the API-consuming app. 

I can't find the existing issue anymore, but I wonder if it's related to this at all: https://github.com/ankidroid/apisample/issues/2

**Comment by vi** [comment]
Note that I didn't reinstall the API sample or TextImporter. It started working after only reinstallation of AnkiDroid.

**Comment by mikehardy** [comment]
Hmm. the combination matrix of all the things to test (which to install first, try to access permission first or not, grant permission from one or the other app or both) is a bit tedious but it would be interesting to know under exactly what conditions it works or fails. I haven't played with the API enough to know it well myself, obviously

**Comment by timrae** [comment]
It sounds like somehow our custom permission that we use for the API had
not been registered in the Android system, causing the apisample to fail
when said permission is requested.

**Comment by vi** [comment]
Then why "Enable API" checkbox in AnkiDroid settings still changed behaviour a bit (delay vs no delay when returning null)
